src/session_handle.cpp

```diff
@@ -760,12 +760,14 @@ namespace {
 
 	unsigned short session_handle::listen_port() const
 	{
-		return sync_call_ret<unsigned short>(&session_impl::listen_port);
+		return sync_call_ret<unsigned short, unsigned short(session_impl::*)() const>
```

Comment:
my assumption here is that the 2nd template argument is necessary to disambiguate an overload, and that the other overload (where you don't specify the interface) is meant to be deprecated. If so, perhaps it would make sense to deprecate the public API that doesn't specify an interface as well. I can't say I've thought it through any more than that.